### PR TITLE
refactor: deprecate tabindex mixin

### PR DESCRIPTION
**Pull request by DevVersion**
Deprecates the `mixinTabIndex`, because it's not very flexible and also always assumes that the given component should fallback to a `tabIndex` of `-1` if disabled. This is a wrong assumption, because
there can be also components that rather would like to remove the `tabIndex` completely using `[attr.tabIndex]`, instead of setting the tabIndex just to `-1`.

For custom elements, removing the `tabIndex` attribute completely means, that those elements are not even programmatically focusable.
For some native elements, like anchors (`<a>`), removing the attribute is wrong, because it would mean that the anchor falls back to a `tabindex` of `0` (_meaning that it's still tabbable_). In that case, we would like to use `-1` instead of removing it.

TL;DR

* Fixes that the `tabindex` attribute is not supported on radio buttons.
* Fixes that some components are still programmatically focusable if disabled (selection-list, select, slider, tab-nav-bar)

**Note**: I decided to not use a host mixin (_as discussed on Slack_) here, because I think it's better to be explicit. It should be clear to what the tabIndex falls back if the component is disabled. Also it's just for a few components, because most of them have an underlying input and therefore don't need a fallback tabindex.

### CODE REVIEW

**Comment by mmalerba** [comment]
Could we not just expand the mixin to take another param? it already has a `defaultTabIndex`, why not a `disabledTabIndex` too?

**Comment by DevVersion** [comment]
@mmalerba Good point. The question would be just: What would be the benefit of using the mixin?

Just having a property, as it's in this PR, feels more clean, readable and also is more explicit. Since we already have to manually parse the `tabindex @Attribute` the mixin is kind of unnecessary, and having an accessor isn't really necessary anymore.

Also the mixin wouldn't know whether the tabindex is being applied through the `[attr.tabindex]` or `[tabIndex]` host binding, and therefore it might result into unexpected behavior.

Being more explicit about the tabindex, since we only need to provide a `disabled` value for ~4 components, feels more right. The other components just redirect the `@Input() tabIndex` to the native underlying input element that handles everything itself.

**Comment by jelbourn** [comment]
Looking it over, I agree that the mixin is not super useful if you need to need to specify the disabled tabIndex as well. The other mixins all do something useful without needing that level of configuration. 

However, we could make it more useful if we could also make it handling reading the tabindex attribute in the mixin constructor to do the initialization. We could use the same requirement (`HasElementRef`) that the color mixin uses to check the native element.

**Comment by DevVersion** [comment]
@jelbourn I see what you mean. I have considered doing this as well, but I don't really see the big benefit here. We would still need to manually inject the `@Attribute()`, and just the `parseInt` would be done by the mixin. In my opinion, that's not worth it. 

In regards to the full mixin (again). I don't like the fact that the mixin doesn't help a lot, and kind of turns the component into something like unpredictable. With that, I mean that everything kind of works magically, and we can't tell immediately how the disabled state is being handled on the component.

**Comment by jelbourn** [comment]
I meant something like `this.elementRef.nativeElement.hasAttribute('tabindex')` to avoid needing `@Attribute`.

**Comment by DevVersion** [comment]
Yeah that works too. I'm fine doing that, but I'd personally prefer keeping such behavior more explicit, because it varies based on the component, and it would be good to keep that logic together. 

We can discuss further, or I can just go with keeping the mixin.

**Comment by jelbourn** [comment]
I'll think about it some more; I'm on the fence now
